pkg/plugins/queue/plugin.go

```diff
@@ -47,6 +47,10 @@ type ReceiveOptions struct {
 	Depth *uint32 `type:"int" required:"false"`
 }
 
+func (p *ReceiveOptions) String() string {
+	return fmt.Sprintf("{QueueName: %s, Depth: %v}", p.QueueName, p.Depth)
+}
+
```

Comment:
These types String() method is used when the service plugin method parameters are being logged in `pkg/adapters/grpc/errors.go`

```
		if len(pe.Args) > 0 {
			args := make(map[string]string)
			for k, v := range pe.Args {
				args[k] = fmt.Sprintf("%v", v)
			}
			ed.Scope.Args = args
		}
```
The intent of using the String() method was to ensure we are logging the parameter values consistently and are not logging any sensitive information, such as content or secrets.

Reply:
Cool that makes perfect sense thanks for pointing it out it's use :+1:. I still think we can do better to improve the quality & maintainability of this code, as manually implementing every `String()` implementation for a struct we need to maintain seems small at first but will continue to grow as we add more structs and plugins.

I think a tag based approach could be more consistent and easier to maintain going forward, and will still meet any security requirements for field privacy, it also makes the purpose much clearer.

```go
type A struct {
  A string `debug:"A"`
  B string `debug:"B"`
  C int `debug:"C"`
}
```
> Note we could use json, but I don't believe it meets the requirement of ignoring fields by omission (no tag), so would not be secure.

Fields not marked with the tag would not be included in the output:
```go
v := reflect.ValueOf(s)

for i := 0; i < v.NumField(); i++ {
    // Get the field tag value
    tag := v.Type().Field(i).Tag.Get("debug")

    // Skip if tag is not defined or ignored
    if tag == "" || tag == "-" {
      continue
    }

    // Append field value to output
}
```

While this is more code than the Stringer methods individually it would be able to support all structs we implement, would have it's own set of test cases, and would greatly reduce refactoring burden if we ever wanted to change the structure of the output.

An alternative could be an established pretty printer like: https://github.com/davecgh/go-spew (I don't like the inclusion of the `unsafe` package though even though we can explicetely exclude it, also it hasn't been maintained in some time.)

Reply:
That sound interesting, and good approach from a security point of view being much more explicit.
For nested structures we would need to walk the graph, but would be a write once utility method hopefully.

Reply:
done